ignite/handlers/time_profilers.py

```diff
@@ -0,0 +1,790 @@
+import functools
+from collections import OrderedDict
+from typing import Any, Callable, Dict, List, Mapping, Sequence, Tuple, Union, cast
+
+import torch
+
+from ignite.engine import Engine, EventEnum, Events
+from ignite.handlers import Timer
+
+
+class BasicTimeProfiler:
+    """
+    BasicTimeProfiler can be used to profile the handlers,
+    events, data loading and data processing times.
+
+    Examples:
+
+    .. code-block:: python
+
+        from ignite.handlers import BasicTimeProfiler
+
+        trainer = Engine(train_updater)
+
+        # Create an object of the profiler and attach an engine to it
+        profiler = BasicTimeProfiler()
+        profiler.attach(trainer)
+
+        @trainer.on(Events.EPOCH_COMPLETED)
+        def log_intermediate_results():
+            profiler.print_results(profiler.get_results())
+
+        trainer.run(dataloader, max_epochs=3)
+
+        profiler.write_results('path_to_dir/time_profiling.csv')
+
```

Comment:
While moving classes from contrib to core, we put versionadded here like : https://github.com/pytorch/ignite/blob/84e51c3b4987c16f6dd27a6abd2f9ff95da3dc6c/ignite/handlers/stores.py#L37
or